hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -198,8 +206,22 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
 
       setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), removedProps, hiveEngineEnabled);
 
-      persistTable(tbl, updateHiveTable);
-      threw = false;
+      try {
+        persistTable(tbl, updateHiveTable);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
+        LOG.error("Cannot tell if commit succeeded, attempting to reconnect and check", persistFailure);
```

Comment:
We might want to add more info about the table to the error message. So it is easier to check the table manually and see what happened based only the logs.

What do you think?

Reply:
I think that is a great idea, I was just working with a multi-tenent application and that kind of info would have been very helpful